packages/core/factories/src/useUserFactory.ts

```diff
@@ -0,0 +1,104 @@
+import { ref, Ref, computed } from '@vue/composition-api';
+import { UseUser } from '@vue-storefront/interfaces';
+
+export type UseUserFactoryParams<USER, UPDATE_USER_PARAMS> = {
+  loadUser: () => Promise<USER>;
+  logOut: (params?: {currentUser?: USER}) => Promise<void>;
+  updateUser: (params: {currentUser: USER; updatedUserData: UPDATE_USER_PARAMS}) => Promise<USER>;
+  register: (params: { email: string; password: string; firstName?: string; lastName?: string }) => Promise<USER>;
+  logIn: (params: { username: string; password: string }) => Promise<USER>;
+  changePassword: (params: {currentUser: USER; currentPassword: string; newPassword: string}) => Promise<USER>;
+};
+
+export function useUserFactory<USER, UPDATE_USER_PARAMS>(
+  factoryParams: UseUserFactoryParams<USER, UPDATE_USER_PARAMS>
+) {
+  const user: Ref<USER> = ref(null);
+  const loading: Ref<boolean> = ref(false);
+  const isAuthenticated = computed(
+    () => user.value && Object.keys(user.value).length > 0
+  );
+
+  return function useUser(): UseUser<USER, UPDATE_USER_PARAMS> {
+    const updateUser = async (params: UPDATE_USER_PARAMS) => {
+      loading.value = true;
+      try {
+        user.value = await factoryParams.updateUser({currentUser: user.value, updatedUserData: params});
+      } catch (err) {
+        throw new Error(err);
+      } finally {
+        loading.value = false;
+      }
+    };
+
+    const register = async (registerUserData: {
+      email: string;
+      password: string;
+      firstName?: string;
+      lastName?: string;
+    }) => {
```

Comment:
needs to be more generic
```suggestion
    const register = async (registerUserData: REGISTER_USER_PARAMS) => {
```

Reply:
Done